aws/aws-parallelcluster-cookbook
# Improve manage_raid resouce code and test

### Description of changes
* Move `mdadm` package installation into `manage_raid` resource
  * Define `:setup` as default action and execute it at install time.
* Rename `manage_raid` custom resource to `raid`
  * Now we have the `:setup` action so it is a more appropriate name.
* Rename head_node_raid recipe to raid_head_node
* Rename compute_raid to raid_compute

### Tests
* Add extra checks in the EBS tests
  * These checks are taken from integration tests (CLI) code.
* Define Inspec test for `mdadm` installation. This package is not installed by default on Centos7 so with the test we verified it is installed.
* Extend raid Inspec tests (test code taken from integration tests)
  * Ensure that the RAID array is reassembled automatically on boot
  * Verify RAID is correctly mounted
* `bash kitchen.ec2.sh environment-config test raid-mounted-centos7`
* `bash kitchen.docker.sh environment-install test raid-centos7`

### References
* https://github.com/aws/aws-parallelcluster/pull/5385
